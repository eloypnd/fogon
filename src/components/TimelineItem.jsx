import React from 'react'
import { Box, Card, Text } from 'rebass'

const TimelineItem = ({id, text}) => (
  <Box width={[1, 1 / 2, 1 / 3]}>
    <Card>
      <Text>{text}</Text>
    </Card>
  </Box>
)

export default TimelineItem
