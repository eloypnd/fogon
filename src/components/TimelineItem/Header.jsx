import React from 'react'
import { Avatar, Box, Flex, Link, Truncate } from 'rebass'

const Header = ({name, profile_image_url_https, screen_name}) => (
  <Flex>
    <Box mr={2}><Avatar src={profile_image_url_https} /></Box>
    <Box>
      <Truncate fontWeight='bold' fontSize={2}>{name}</Truncate>
      <Link href={`https://twitter.com/${screen_name}`} target='_blank'>
        @{screen_name}
      </Link>
    </Box>
  </Flex>
)

export default Header
