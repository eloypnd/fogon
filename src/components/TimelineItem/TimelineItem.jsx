import React from 'react'
import Header from './Header'
import { Box, Card, Image, Text } from 'rebass'

const TimelineItem = ({media, message, user}) => (
  <Box width={[1, 1 / 2, 1 / 3]} p={1}>
    <Card>
      <Header {...user} />
      {media && media[0].type === 'photo' &&
        <Image src={media[0].media_url_https} my={2} />
      }
      <Text
        dangerouslySetInnerHTML={{__html: message}}
      />
    </Card>
  </Box>
)

export default TimelineItem
