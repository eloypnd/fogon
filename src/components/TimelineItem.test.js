import React from 'react'
import renderer from 'react-test-renderer'
import TimelineItem from './TimelineItem'

it('TimelineItem renders correctly', () => {
  const tree = renderer
    .create(<TimelineItem />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
