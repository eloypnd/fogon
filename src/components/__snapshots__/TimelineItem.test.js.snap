// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem renders correctly 1`] = `
<div
  className="sc-bdVaJa dbOKEe"
>
  <div
    className="sc-jAaTju hdyfev"
  >
    <div
      className="sc-gqjmRU kKsieB"
    />
  </div>
</div>
`;
